### PR TITLE
bpo-30858: Improve error location for expressions with assignments

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue30858

### CODE REVIEW

**Comment by pablogsal** [comment]
@lysnikolaou I was in automatic mode and I realized that I opened the PR without considering if you would have preferred to open it yourself. Please, feel free to close this and open one yourself if you prefer so :)

File: Grammar/python.gram [hunk 1]
```diff
@@ -646,9 +646,9 @@ invalid_arguments:
         RAISE_SYNTAX_ERROR_KNOWN_LOCATION(a, "Generator expression must be parenthesized") }
     | a=args ',' args { _PyPegen_arguments_parsing_error(p, a) }
 invalid_kwarg:
-    | a=expression '=' {
+    | a=expression b='=' {
```
Comment (lysnikolaou) [inline Grammar/python.gram]: Let's not have an unused variable here.

```suggestion
    | expression a='=' {
```

**Review by lysnikolaou — COMMENTED** [review]
No wories at all @pablogsal!

**Comment by lysnikolaou** [comment]
I think there's also a test that needs changing in `test_exceptions`, `testSyntaxErrorOffset`.

**Review by lysnikolaou — APPROVED** [review]
Yup!

**Pull request merged**